block/provider/list_snapshots.go

```diff
@@ -18,13 +18,76 @@
 package provider
 
 import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM/ibmcloud-volume-interface/lib/metrics"
 	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	userError "github.com/IBM/ibmcloud-volume-vpc/common/messages"
+	"github.com/IBM/ibmcloud-volume-vpc/common/vpcclient/models"
+	"go.uber.org/zap"
 )
 
+const startSnapshoIDNotFoundMsg = "start parameter is not valid"
+
 // ListSnapshots list all snapshots
-func (vpcs *VPCSession) ListSnapshots() ([]*provider.Snapshot, error) {
+func (vpcs *VPCSession) ListSnapshots(limit int, start string, filters map[string]string) (*provider.SnapshotList, error) {
 	vpcs.Logger.Info("Entry ListeSnapshots")
 	defer vpcs.Logger.Info("Exit ListSnapshots")
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "ListSnapshots", time.Now())
+
+	if limit < 0 {
+		return nil, userError.GetUserError("InvalidListSnapshotLimit", nil, limit)
+	}
+
+	if limit > maxLimit {
```

Comment:
This looks to be bit odd, is this not verified by vpc ?

Reply:
`maxLimit                 = 100`
is set in our code as we cannot handle all the response that vpc sends. Hence this checks needs to be here